src/common/extensionMessaging.ts

```diff
@@ -35,7 +47,8 @@ export class ExtensionMessageSender implements IExtensionMessageSender {
 
     public sendMessage(message: ExtensionMessage, args?: any[]): Q.Promise<any> {
         let deferred = Q.defer<any>();
-        let messageWithArguments: MessageWithArguments = { message: message, args: args };
+        let messageWithArguments: MessageWithArguments = { message: message, args: args || [] };
```

Comment:
Why are we changing the protocol here to send an empty array if the caller doesn't provide args?